projects/plugins/boost/app/modules/optimizations/lcp/class-lcp.php

```diff
@@ -2,22 +2,205 @@
+		if ( is_customize_preview() ) {
+			return;
+		}
+
+		// Disable in feeds, AJAX, Cron, XML.
+		if ( is_feed() || wp_doing_ajax() || wp_doing_cron() || wp_is_xml_request() ) {
+			return;
+		}
+
+		// Disable in sitemaps.
+		if ( ! empty( get_query_var( 'sitemap' ) ) ) {
+			return;
+		}
+
+		// Disable in AMP pages.
+		if ( function_exists( 'amp_is_request' ) && amp_is_request() ) {
+			return;
+		}
+
+		$this->output_filter->add_callback( array( $this, 'optimize' ) );
+	}
+
+	/**
+	 * Optimize the HTML content by finding the LCP image and adding required attributes.
+	 *
+	 * @param string $buffer_start First part of the buffer.
+	 * @param string $buffer_end   Second part of the buffer.
+	 *
+	 * @return array Parts of the buffer.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function optimize( $buffer_start, $buffer_end ) {
+		// Get the LCP image tag from WP option
+		$lcp_image_tag = $this->get_lcp_image_tag();
+
+		// Early return if no tag is configured
+		if ( empty( $lcp_image_tag ) ) {
+			return array( $buffer_start, $buffer_end );
+		}
+
+		// Combine the buffers for processing
+		$combined_buffer = $buffer_start . $buffer_end;
+
+		// Check if the tag exists in the combined buffer
+		if ( strpos( $combined_buffer, $lcp_image_tag ) === false ) {
+			return array( $buffer_start, $buffer_end );
+		}
+
+		// Create the optimized tag with required attributes
+		$optimized_tag = $this->optimize_image_tag( $lcp_image_tag );
+
+		// If no optimization was needed, return early
+		if ( $optimized_tag === $lcp_image_tag ) {
+			return array( $buffer_start, $buffer_end );
+		}
+
+		// Simple string replacement (since we're looking for an exact tag)
+		$modified_buffer = str_replace( $lcp_image_tag, $optimized_tag, $combined_buffer );
+
+		// Split the modified buffer back into two parts
+		$buffer_start_length = strlen( $buffer_start );
+		$new_buffer_start    = substr( $modified_buffer, 0, $buffer_start_length );
+		$new_buffer_end      = substr( $modified_buffer, $buffer_start_length );
+
+		// Check for successful split
+		if ( false === $new_buffer_start || false === $new_buffer_end ) {
+			// If splitting failed, return the original buffers
+			return array( $buffer_start, $buffer_end );
+		}
+
+		return array( $new_buffer_start, $new_buffer_end );
+	}
+
+	/**
+	 * Optimize an image tag by adding required attributes.
+	 *
+	 * @param string $tag The original image tag.
+	 * @return string The optimized image tag.
+	 *
+	 * @since $$next-version$$
+	 */
+	private function optimize_image_tag( $tag ) {
+		// Add fetchpriority="high" if not present
+		if ( ! preg_match( '/fetchpriority\s*=\s*["\']high["\']/i', $tag ) ) {
+			$tag = preg_replace( '/<img\s/i', '<img fetchpriority="high" ', $tag );
+		}
+
+		// Add loading="eager" if not present
+		if ( ! preg_match( '/loading\s*=\s*["\']eager["\']/i', $tag ) ) {
+			$tag = preg_replace( '/<img\s/i', '<img loading="eager" ', $tag );
+		}
+
+		return $tag;
+	}
```

Comment:
I think we discussed it in the team call, but what would you think about moving to the WP HTML API for this work, just like we've done in #32700? It may be more future proof.